mmtf-codec/src/test/java/org/rcsb/mmtf/serialization/mp/ZippedTextFile.java

```diff
@@ -0,0 +1,29 @@
+package org.rcsb.mmtf.serialization.mp;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipInputStream;
+
+public class ZippedTextFile {
+
+	public static List<String> readLines(File f) throws IOException {
+		List<String> lines = new ArrayList<>();
+		String line;
+		try (ZipInputStream zis = new ZipInputStream(new FileInputStream(f))) {
+			// set the position in stream to the first and only file
+			zis.getNextEntry();
+			BufferedReader br = new BufferedReader(new InputStreamReader(zis));
+			while ((line = br.readLine()) != null) {
+				lines.add(line);
+			}
+			zis.closeEntry();
+			return lines;
+		}
+	}
+
+}

```

Comment:
You need gzipping rather than zipping here.

This is how you do it with gzipping:
```java
public static void gzipFile(File inFile, File outFile) throws IOException {
		GZIPOutputStream zos = new GZIPOutputStream(new FileOutputStream(outFile));
		FileInputStream is = new FileInputStream(inFile);

		int b;
		while ( (b=is.read())!=-1) {
			zos.write(b);
		}
		zos.close();
		is.close();
	}
```
And ungzip is:
```java
public static void gunzipFile(File inFile, File outFile) throws IOException {
		GZIPInputStream zis = new GZIPInputStream(new FileInputStream(inFile));
		FileOutputStream os = new FileOutputStream(outFile);
		int b;
		while ( (b=zis.read())!=-1) {
			os.write(b);
		}
		zis.close();
		os.close();
	}
```